src/Group.php

```diff
@@ -2,16 +2,18 @@
 
 namespace Yiisoft\Router;
 
+use InvalidArgumentException;
 use Psr\Http\Server\MiddlewareInterface;
 use Yiisoft\Router\Middleware\Callback;
+use function is_callable;
 
 class Group implements RouteCollectorInterface
 {
-    protected $items = [];
-    protected $prefix;
-    protected $middlewares = [];
+    protected array $items = [];
+    protected ?string $prefix;
+    protected array $middlewares = [];
 
-    public function __construct(string $prefix = null, callable $callback = null)
+    public function __construct(?string $prefix = null, ?callable $callback = null)
```

Comment:
There's no need making these nullable.

Reply:
I'm not agree with you. I can extend this class in my project and put given parameters (including default values) to the parent:
```php
    public function __construct(?string $prefix = null, ?callable $callback = null)
    {
        // my stuff
        parent::__construct($prefix, $callback);
    }
```

Reply:
OK. That's valid.